addon/wolfcrypt/ecc.cpp

```diff
@@ -0,0 +1,217 @@
+{
+  int ret;
+  Napi::Env env = info.Env();
+  ecc_key* ecc = (ecc_key*)( info[0].As<Napi::Uint8Array>().Data() );
+
+  ret = wc_ecc_size( ecc );
+
+  return Napi::Number::New( env, ret );
+}
+
+Napi::Number bind_wc_ecc_init(const Napi::CallbackInfo& info)
+{
+  int ret;
+  Napi::Env env = info.Env();
+  ecc_key* ecc = (ecc_key*)( info[0].As<Napi::Uint8Array>().Data() );
+
+  ecc->rng = NULL;
+  ret = wc_ecc_init( ecc );
+
+  return Napi::Number::New( env, ret );
+}
+
+Napi::Number bind_wc_ecc_make_key(const Napi::CallbackInfo& info)
+{
+  Napi::Env env = info.Env();
+  int ret;
+  int key_size = info[0].As<Napi::Number>().Int32Value();
+  ecc_key* ecc = (ecc_key*)( info[1].As<Napi::Uint8Array>().Data() );
+
+  ecc->rng = wc_rng_new( NULL, 0, NULL );
+
+  ret = wc_ecc_make_key( ecc->rng, key_size, ecc );
+
+  return Napi::Number::New( env, ret );
+}
+
+Napi::Number sizeof_ecc_x963(const Napi::CallbackInfo& info)
+{
+  Napi::Env env = info.Env();
+  ecc_key* ecc = (ecc_key*)( info[0].As<Napi::Uint8Array>().Data() );
+  unsigned int out_len;
+
+  wc_ecc_export_x963( ecc, NULL, &out_len );
+
+  return Napi::Number::New( env, out_len );
+}
+
+Napi::Number bind_wc_ecc_export_x963(const Napi::CallbackInfo& info)
+{
+  Napi::Env env = info.Env();
+  int ret;
+  ecc_key* ecc = (ecc_key*)( info[0].As<Napi::Uint8Array>().Data() );
+  uint8_t* out = (uint8_t*)( info[1].As<Napi::Uint8Array>().Data() );
+  unsigned int out_len = info[2].As<Napi::Number>().Int32Value();
+
+  ret = wc_ecc_export_x963( ecc, out, &out_len );
+
+  if ( ret < 0 )
+  {
+    out_len = ret;
+  }
+
+  return Napi::Number::New( env, (int)out_len );
+}
+
+Napi::Number bind_wc_ecc_import_x963(const Napi::CallbackInfo& info)
+{
+  Napi::Env env = info.Env();
+  int ret;
+  uint8_t* in = (uint8_t*)( info[0].As<Napi::Uint8Array>().Data() );
+  unsigned int in_len = info[1].As<Napi::Number>().Int32Value();
+  ecc_key* ecc = (ecc_key*)( info[2].As<Napi::Uint8Array>().Data() );
+
+  ret = wc_ecc_import_x963( in, in_len, ecc );
+
+  return Napi::Number::New( env, ret );
+}
+
+Napi::Number bind_wc_ecc_set_curve(const Napi::CallbackInfo& info)
+{
+  Napi::Env env = info.Env();
+  int ret;
+  ecc_key* ecc = (ecc_key*)( info[0].As<Napi::Uint8Array>().Data() );
+  int key_size = info[1].As<Napi::Number>().Int32Value();
+  int curve_id = info[2].As<Napi::Number>().Int32Value();
+
+  ret = wc_ecc_set_curve( ecc, key_size, curve_id );
+
+  return Napi::Number::New( env, ret );
+}
+
+Napi::Number bind_wc_ecc_shared_secret(const Napi::CallbackInfo& info)
+{
+  Napi::Env env = info.Env();
+  int ret;
+  ecc_key* private_key = (ecc_key*)( info[0].As<Napi::Uint8Array>().Data() );
+  ecc_key* public_key = (ecc_key*)( info[1].As<Napi::Uint8Array>().Data() );
+  uint8_t* out = info[2].As<Napi::Uint8Array>().Data();
+  unsigned int out_len = info[3].As<Napi::Number>().Uint32Value();
+
+  ret = wc_ecc_shared_secret( private_key, public_key, out, &out_len );
```

Comment:
In a few cases you will need to set `wc_ecc_set_rng` here. Specifically when `ECC_TIMING_RESISTANT` is defined.